cardano-api/src/Cardano/Api/SpecialByron.hs

```diff
@@ -0,0 +1,199 @@
+{-# LANGUAGE TypeFamilies #-}
+
+-- | Special Byron values that we can submit to a node to propose an update proposal
+-- or to vote on an update proposal. These are not transactions.
+--
+module Cardano.Api.SpecialByron
+  ( ByronUpdateProposal(..),
+    ByronProtocolParametersUpdate(..),
+    AsType(AsByronUpdateProposal, AsByronVote),
+    updateProposalToGenTx,
+    createProtocolParametersUpdate,
+    ByronVote(..),
+    createByronUpdateProposal,
+    createByronVote,
+    voteToGenTx,
```

Comment:
Naming:

For whatever reason, the naming convention elsewhere in the API for these kinds of exported construction functions is "makeBlah". We should probably stick to that.

For conversions, the naming convention elsewhere is "toByronBlah". I can see that "toByronUpdateProposal" and "toByronVote" might seem a bit awkward since they would convert from types with those names, even though they would convert to Byron ledger types.

How about?
* `toByronLedgerUpdateProposal`
* `toByronLedgertoByronVote`